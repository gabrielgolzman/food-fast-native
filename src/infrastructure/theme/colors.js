export const colors = {
   ui: {
      primary: '#F3A404',
      secondary: '#F1062E',
      bg: '#FFFFFF',
   },
};
