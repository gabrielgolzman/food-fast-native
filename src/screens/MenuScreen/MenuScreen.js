import React from 'react';
import { View, TouchableNativeFeedback } from 'react-native';

import { MenuList } from './MenuScreen.styles';

import ProductCard from '../../components/restaurant/ProductCard/ProductCard';
import { FadeInView } from '../../utility/animations/FadeAnimation';

const MenuScreen = () => {
   let mock = require('../../../data/mock.json');
   return (
      <MenuList
         data={mock.menuOptions}
         renderItem={({ item }) => {
            return (
               <TouchableNativeFeedback
                  onPress={() => {
                     console.log(item.optionName + ' selected!');
                  }}
                  background={TouchableNativeFeedback.Ripple('#FFFFFF', true)}
               >
                  <View>
                     <FadeInView>
                        <ProductCard product={item} />
                     </FadeInView>
                  </View>
               </TouchableNativeFeedback>
            );
         }}
         keyExtractor={(item) => item.optionName}
      />
   );
};

export default MenuScreen;
