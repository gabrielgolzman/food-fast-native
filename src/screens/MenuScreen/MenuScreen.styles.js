import styled from 'styled-components/native';

export const MenuList = styled.FlatList.attrs({
   contentContainerStyle: {
      padding: 16,
   },
})``;
