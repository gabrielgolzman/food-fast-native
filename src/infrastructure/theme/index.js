import { colors } from './colors';
import { space } from './spacing';
import { fontSizes, fontWeights } from './fonts';

export const theme = { colors, space, fontSizes, fontWeights };
