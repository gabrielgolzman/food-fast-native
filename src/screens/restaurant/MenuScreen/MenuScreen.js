import React, { useContext, useEffect } from 'react';
import { View, TouchableNativeFeedback } from 'react-native';

import { MenuList } from './MenuScreen.styles';
import { OrdersContext } from '../../../services/orders/orders.context';

import ProductCard from '../../../components/restaurant/ProductCard/ProductCard';
import Layout from '../../../utility/Layout/Layout';
import { FadeInView } from '../../../utility/animations/FadeAnimation';
import Button from '../../../components/ui/Button/Button';

const MenuScreen = ({ navigation }) => {
   let mock = require('../../../../data/mock.json');

   const { orders, total, setOrderTotal } = useContext(OrdersContext);

   useEffect(() => {
      if (orders.length !== 0) {
         const amountArray = orders.map((order) => {
            return order.partialPrice;
         });
         const amount = amountArray.reduce((cur, val) => cur + val);
         setOrderTotal(amount);
      }
   }, [orders]);

   const orderButton = total ? (
      <Button onPress={() => navigation.navigate('summary')} type="sticky">
         Ver mi pedido ${+total.toFixed(2)}
      </Button>
   ) : null;

   return (
      <Layout nav={navigation}>
         <MenuList
            data={mock.menuOptions}
            renderItem={({ item }) => {
               return (
                  <TouchableNativeFeedback
                     onPress={() => {
                        navigation.navigate('product', { product: item });
                     }}
                     background={TouchableNativeFeedback.Ripple(
                        '#FFFFFF',
                        true
                     )}
                  >
                     <View>
                        <FadeInView>
                           <ProductCard product={item} />
                        </FadeInView>
                     </View>
                  </TouchableNativeFeedback>
               );
            }}
            keyExtractor={(item) => item.optionName}
         />
         {orderButton}
      </Layout>
   );
};

export default MenuScreen;
