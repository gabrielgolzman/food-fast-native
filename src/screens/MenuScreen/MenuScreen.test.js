import React from 'react';
import { render } from '@testing-library/react-native';

import MenuScreen from './MenuScreen';
import { theme } from '../../infrastructure/theme/index';
import { ThemeProvider } from 'styled-components/native';

it('renders the MenuScreen correctly', () => {
   const tree = render(
      <ThemeProvider theme={theme}>
         <MenuScreen />
      </ThemeProvider>
   ).toJSON();
   expect(tree).toMatchSnapshot();
});
