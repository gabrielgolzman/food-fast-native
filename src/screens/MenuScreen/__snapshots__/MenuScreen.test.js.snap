// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the MenuScreen correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexDirection": "column-reverse",
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
          "height": "30%",
          "justifyContent": "space-between",
          "position": "absolute",
          "top": 0,
          "width": "100%",
          "zIndex": 1,
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 6,
          "left": 6,
          "right": 6,
          "top": 6,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "margin": 6,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 26.25,
              "height": 52.5,
              "width": 52.5,
            },
            undefined,
            Array [
              Object {
                "zIndex": 10,
              },
            ],
          ],
        ]
      }
    >
      <View>
        <Text />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "orange",
          "borderColor": "transparent",
          "borderRadius": 20,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "fontSize": 12,
          "height": 40,
          "justifyContent": "center",
          "marginBottom": 8,
          "marginLeft": 8,
          "marginRight": 8,
          "marginTop": 8,
          "minWidth": 64,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
          "width": "70%",
          "zIndex": 10,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 20,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "marginLeft": 12,
                "marginRight": -4,
              }
            }
          >
            <Text />
          </View>
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "color": "white",
                    "fontSize": 12,
                  },
                ],
              ]
            }
          >
            Solicitar ayuda al personal
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "height": "70%",
        },
      ]
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "padding": 16,
        }
      }
      data={
        Array [
          Object {
            "category": "café",
            "dateAvailableFrom": "2021-01-01",
            "dateAvailableUntil": "2021-06-30",
            "description": "El mejor café colombiano que probarás jamás",
            "idMenuOption": 23,
            "isAvailable": true,
            "optionName": "Café colombiano",
            "optionNumber": 1,
            "photo": "https://www.mdzol.com/u/fotografias/m/2021/4/8/f850x638-1041350_1118839_5050.jpg",
            "unitPrice": 150,
          },
          Object {
            "category": "almuerzos",
            "dateAvailableFrom": "2021-10-25",
            "dateAvailableUntil": "2021-07-01",
            "description": "Minuta ejecutiva típica con bebida a elección",
            "idMenuOption": 29,
            "isAvailable": true,
            "optionName": "Milanesa completa con bebida",
            "optionNumber": 2,
            "photo": "https://cdn.needish.com/is-prod-deals/aiY6rmMwouS2sSt7-qEFDg/scale/900x600.jpg",
            "unitPrice": 1450,
          },
          Object {
            "category": "café",
            "dateAvailableFrom": "2021-06-24",
            "dateAvailableUntil": "2021-07-02",
            "description": "Café típico con un tercio de café y resto de leche",
            "idMenuOption": 17,
            "isAvailable": true,
            "optionName": "Lágrima",
            "optionNumber": 3,
            "photo": "https://i.pinimg.com/originals/91/1a/fe/911afeef717a9325ed8965b33ea897e2.jpg",
            "unitPrice": 175.49,
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Array [
          Object {},
        ]
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View
              style={
                Object {
                  "opacity": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "rgba(0, 0, 0, 0.12)",
                    "borderRadius": 4,
                    "elevation": 5,
                    "marginBottom": 16,
                    "shadowColor": "#000000",
                    "shadowOffset": Object {
                      "height": 4,
                      "width": 0,
                    },
                    "shadowOpacity": 0.24,
                    "shadowRadius": 5,
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flexGrow": 1,
                      "flexShrink": 1,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#eeeeee",
                          "height": 195,
                          "overflow": "hidden",
                        },
                        Object {
                          "borderTopLeftRadius": 4,
                          "borderTopRightRadius": 4,
                        },
                        Array [
                          Object {
                            "backgroundColor": "#FFFFFF",
                            "paddingBottom": 8,
                            "paddingLeft": 8,
                            "paddingRight": 8,
                            "paddingTop": 8,
                          },
                        ],
                      ]
                    }
                  >
                    <Image
                      alt="Café colombiano"
                      siblings={
                        Array [
                          "Styled(withTheme(Card.Cover))",
                          "Styled(View)",
                        ]
                      }
                      source={
                        Object {
                          "uri": "https://www.mdzol.com/u/fotografias/m/2021/4/8/f850x638-1041350_1118839_5050.jpg",
                        }
                      }
                      style={
                        Array [
                          Object {
                            "flex": 1,
                            "height": undefined,
                            "justifyContent": "flex-end",
                            "padding": 16,
                            "width": undefined,
                          },
                          Object {
                            "borderTopLeftRadius": 4,
                            "borderTopRightRadius": 4,
                          },
                        ]
                      }
                    />
                  </View>
                  <View
                    index={1}
                    siblings={
                      Array [
                        "Styled(withTheme(Card.Cover))",
                        "Styled(View)",
                      ]
                    }
                    style={
                      Array [
                        Object {
                          "paddingBottom": 16,
                          "paddingLeft": 16,
                          "paddingRight": 16,
                          "paddingTop": 16,
                        },
                      ]
                    }
                    total={2}
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "fontSize": 24,
                            "marginBottom": 0,
                          },
                        ]
                      }
                    >
                      Café colombiano
                    </Text>
                    <Text
                      style={
                        Array [
                          Object {
                            "fontSize": 12,
                            "marginBottom": 8,
                          },
                        ]
                      }
                    >
                      El mejor café colombiano que probarás jamás
                    </Text>
                    <Text
                      style={
                        Array [
                          Object {
                            "fontSize": 20,
                            "fontWeight": "700",
                          },
                        ]
                      }
                    >
                      $
                      150
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View
              style={
                Object {
                  "opacity": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "rgba(0, 0, 0, 0.12)",
                    "borderRadius": 4,
                    "elevation": 5,
                    "marginBottom": 16,
                    "shadowColor": "#000000",
                    "shadowOffset": Object {
                      "height": 4,
                      "width": 0,
                    },
                    "shadowOpacity": 0.24,
                    "shadowRadius": 5,
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flexGrow": 1,
                      "flexShrink": 1,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#eeeeee",
                          "height": 195,
                          "overflow": "hidden",
                        },
                        Object {
                          "borderTopLeftRadius": 4,
                          "borderTopRightRadius": 4,
                        },
                        Array [
                          Object {
                            "backgroundColor": "#FFFFFF",
                            "paddingBottom": 8,
                            "paddingLeft": 8,
                            "paddingRight": 8,
                            "paddingTop": 8,
                          },
                        ],
                      ]
                    }
                  >
                    <Image
                      alt="Milanesa completa con bebida"
                      siblings={
                        Array [
                          "Styled(withTheme(Card.Cover))",
                          "Styled(View)",
                        ]
                      }
                      source={
                        Object {
                          "uri": "https://cdn.needish.com/is-prod-deals/aiY6rmMwouS2sSt7-qEFDg/scale/900x600.jpg",
                        }
                      }
                      style={
                        Array [
                          Object {
                            "flex": 1,
                            "height": undefined,
                            "justifyContent": "flex-end",
                            "padding": 16,
                            "width": undefined,
                          },
                          Object {
                            "borderTopLeftRadius": 4,
                            "borderTopRightRadius": 4,
                          },
                        ]
                      }
                    />
                  </View>
                  <View
                    index={1}
                    siblings={
                      Array [
                        "Styled(withTheme(Card.Cover))",
                        "Styled(View)",
                      ]
                    }
                    style={
                      Array [
                        Object {
                          "paddingBottom": 16,
                          "paddingLeft": 16,
                          "paddingRight": 16,
                          "paddingTop": 16,
                        },
                      ]
                    }
                    total={2}
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "fontSize": 24,
                            "marginBottom": 0,
                          },
                        ]
                      }
                    >
                      Milanesa completa con bebida
                    </Text>
                    <Text
                      style={
                        Array [
                          Object {
                            "fontSize": 12,
                            "marginBottom": 8,
                          },
                        ]
                      }
                    >
                      Minuta ejecutiva típica con bebida a elección
                    </Text>
                    <Text
                      style={
                        Array [
                          Object {
                            "fontSize": 20,
                            "fontWeight": "700",
                          },
                        ]
                      }
                    >
                      $
                      1450
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View
              style={
                Object {
                  "opacity": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "rgba(0, 0, 0, 0.12)",
                    "borderRadius": 4,
                    "elevation": 5,
                    "marginBottom": 16,
                    "shadowColor": "#000000",
                    "shadowOffset": Object {
                      "height": 4,
                      "width": 0,
                    },
                    "shadowOpacity": 0.24,
                    "shadowRadius": 5,
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flexGrow": 1,
                      "flexShrink": 1,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#eeeeee",
                          "height": 195,
                          "overflow": "hidden",
                        },
                        Object {
                          "borderTopLeftRadius": 4,
                          "borderTopRightRadius": 4,
                        },
                        Array [
                          Object {
                            "backgroundColor": "#FFFFFF",
                            "paddingBottom": 8,
                            "paddingLeft": 8,
                            "paddingRight": 8,
                            "paddingTop": 8,
                          },
                        ],
                      ]
                    }
                  >
                    <Image
                      alt="Lágrima"
                      siblings={
                        Array [
                          "Styled(withTheme(Card.Cover))",
                          "Styled(View)",
                        ]
                      }
                      source={
                        Object {
                          "uri": "https://i.pinimg.com/originals/91/1a/fe/911afeef717a9325ed8965b33ea897e2.jpg",
                        }
                      }
                      style={
                        Array [
                          Object {
                            "flex": 1,
                            "height": undefined,
                            "justifyContent": "flex-end",
                            "padding": 16,
                            "width": undefined,
                          },
                          Object {
                            "borderTopLeftRadius": 4,
                            "borderTopRightRadius": 4,
                          },
                        ]
                      }
                    />
                  </View>
                  <View
                    index={1}
                    siblings={
                      Array [
                        "Styled(withTheme(Card.Cover))",
                        "Styled(View)",
                      ]
                    }
                    style={
                      Array [
                        Object {
                          "paddingBottom": 16,
                          "paddingLeft": 16,
                          "paddingRight": 16,
                          "paddingTop": 16,
                        },
                      ]
                    }
                    total={2}
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "fontSize": 24,
                            "marginBottom": 0,
                          },
                        ]
                      }
                    >
                      Lágrima
                    </Text>
                    <Text
                      style={
                        Array [
                          Object {
                            "fontSize": 12,
                            "marginBottom": 8,
                          },
                        ]
                      }
                    >
                      Café típico con un tercio de café y resto de leche
                    </Text>
                    <Text
                      style={
                        Array [
                          Object {
                            "fontSize": 20,
                            "fontWeight": "700",
                          },
                        ]
                      }
                    >
                      $
                      175.49
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
